documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,311 @@
+# Setting Up Security for a Distributed Mode Cluster
+
+In the Distributed Mode of running a cluster, each component runs separately on one or more processes, usually distributed across multiple machines. The deployment options in this mode include:
+
+* A manual deployment in hardware or virtual machines
+* Containerized deployments of these types:
+  * A Kubernetes native application deployed using the Pravega Operator
+  * A Docker Compose application deployment
+  * A Docker Swarm based distributed deployment
+
+Regardless of the deployment option used, setting up Transport Layer Security (SSL/TLS) for as well as client authentication and authorization (`Auth` in short) are important steps towards a secure deployment. TLS enables encryption of client-server and inter-component traffic and the clients to authenticate the server. Client `auth` enables the servers to authenticate and authorize the clients. Pravega strongly recommends always enabling security for production clusters.
+
+Setting up security - especially TLS - in a large cluster can be daunting at first. To make it easier, this document provides step-by-step instructions on how to enable security when deloying a Pravega cluster manually.
+
+Depending on the deployment option used and your environment, you might need to modify the steps and commands  to suite your specific needs and policies. Also, some of the deployment options open up additional ways of securing a cluster, as we will see in the next sub-section; We'll not discuss those ways in detail in this document, for the sake of brevity.
+
+## Setting up SSL/TLS
+
+There are two common ways of using TLS for client-server communications:
+
+1. Setup Pravega to handle TLS directly.
+2. Terminate TLS outside of Pravega, in an infrastructure component such as a reverse proxy or a Load Balancer.
+
+Depending on the deployment option used, it might be easier to use one or the other approach. For instance, if you are deploying a Pravega cluster in Kubernetes, you might find using approach 2 simpler to use and manage. Also, the specifics of how to enable TLS will differ depending on the deployment option used. This document focuses on approach 1. For approach 2, refer to the vendor's documentation.
+
+At a high level, setting up TLS in Pravega can be divided into three distinct stages:
+
```

Comment:
instead stage use phase

Reply:
The reason I used stage instead of phase is phase usually conjures up an image of stages where once you go forward you don't come back, and has a more formal tone. I prefer to use the word stage in this context.